### PR TITLE
Buckets need to be explicitly created - no more auto creation

### PR DESCRIPTION
**Pull request by DeepDiver1975**
- adding s3:createBucket command
- autocreate options has been removed
- list CORS settings in ls command - needed for future releases

### CODE REVIEW

File: lib/command/createbucket.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	public function __construct(IConfig $config) {
+		parent::__construct();
+		$this->config = $config;
+	}
+
+	protected function configure() {
+		$this
+			->setName('s3:create-bucket')
+			->setDescription('Create a bucket as necessary to be used')
+			->addArgument('bucket', InputArgument::REQUIRED, 'Name of the bucket; it`s objects will be listed')
```
Comment (PVince81) [inline lib/command/createbucket.php]: => its objects

File: lib/command/createbucket.php [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+			if (!$helper->ask($input, $output, new ConfirmationQuestion($q, false))) {
+				return 1;
+			}
+		}
+
+		$bucketName = $input->getArgument('bucket');
+		$client = $this->getClient();
+		$result = $client->doesBucketExist($bucketName);
+		if ($result) {
+			$output->writeln("Bucket does already exist: $bucketName");
```
Comment (PVince81) [inline lib/command/createbucket.php]: => "Bucket already exists"

File: lib/command/createbucket.php [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+			->addArgument('bucket', InputArgument::REQUIRED, 'Name of the bucket; it`s objects will be listed')
+			->addOption('update-configuration', null, InputOption::VALUE_NONE, 'If the bucket exists the configuration will be updated')
+			->addOption('accept-warning', null, InputOption::VALUE_NONE, 'No warning about the usage of this command will be displayed');
+	}
+
+	protected function execute(InputInterface $input, OutputInterface $output) {
+		if (!$input->getOption('accept-warning')) {
+			$helper = new QuestionHelper();
+			$q = <<<EOS
+<question>This command is mainly for development purpose.
```
Comment (PVince81) [inline lib/command/createbucket.php]: "development purposes"

File: lib/command/createbucket.php [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+	public function __construct(IConfig $config) {
+		parent::__construct();
+		$this->config = $config;
+	}
+
+	protected function configure() {
+		$this
+			->setName('s3:create-bucket')
+			->setDescription('Create a bucket as necessary to be used')
+			->addArgument('bucket', InputArgument::REQUIRED, 'Name of the bucket; its objects will be listed')
```
Comment (patrickjahns) [inline lib/command/createbucket.php]: can you clarify - I don't see code here listing the objects

**Comment by patrickjahns** [comment]
Could you clarify why the changes to `ls` command ?

**Comment by DeepDiver1975** [comment]
> Could you clarify why the changes to ls command ?

1. wrong license header was used in the file - the whole repo was GPL since the first day
2. listing CORS parameters will be of interest in future release - would be good to allow admins to list them today already

File: appinfo/info.xml [hunk 5]
```diff
@@ -31,6 +31,7 @@ This extension is the successor of the [ownCloud Objectstore App](https://market
 	</types>
 	<commands>
 		<command>OCA\Files_Primary_S3\Command\ls</command>
+		<command>OCA\Files_Primary_S3\Command\createBucket</command>
```
Comment (butonic) [inline appinfo/info.xml]: why lowercase?!? It is a gooddamn class

Comment (DeepDiver1975) [reply appinfo/info.xml]: this was asked and responded to in the other pr already.

the app does not yet follow the OCA namespace semantics - actually the objectstore loading code in core is not respecting it. This needs a debugging session of it's own

**Comment by PVince81** [comment]
are we ok with merging in the light of the recent clarifications ? @patrickjahns @butonic 

(once CI is done)

**Pull request merged**